kernels/optimized/cpu/op_add_sub_impl.h

```diff
@@ -0,0 +1,219 @@
+    // support for - operator. At least not right now.
+    ScalarType out_type = out.scalar_type();
+    ET_SWITCH_REALB_TYPES(out_type, ctx, op_name, CTYPE, [&]() {
+      CTYPE alpha_val;
+      ET_KERNEL_CHECK_MSG(
+          ctx,
+          torch::executor::native::utils::extract_scalar(alpha, &alpha_val),
+          InvalidArgument,
+          out,
+          "Failed to extract scalar alpha.");
+      using Vec = executorch::vec::Vectorized<CTYPE>;
+      Vec alpha_val_vec(alpha_val);
+      if constexpr (is_sub) {
+        if (selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcast2dBy1dReverseArguments ||
+            selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcastLastDimReverseArguments ||
+            selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcastNdByNdReverseArguments) {
+          auto add_lambda = [&alpha_val_vec](auto x, auto y) {
+            return y - alpha_val_vec * x;
+          };
+          return torch::executor::handle_broadcast_elementwise<CTYPE>(
+              ctx, add_lambda, a, b, out, selected_optimized_path, alpha);
+        } else {
+          auto add_lambda = [&alpha_val_vec](auto x, auto y) {
+            return x - alpha_val_vec * y;
+          };
+          return torch::executor::handle_broadcast_elementwise<CTYPE>(
+              ctx, add_lambda, a, b, out, selected_optimized_path, alpha);
+        }
+      } else {
+        if (selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcast2dBy1dReverseArguments ||
+            selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcastLastDimReverseArguments ||
+            selected_optimized_path ==
+                ElementwiseOptimizedPath::kBroadcastNdByNdReverseArguments) {
+          // Reason we swap out args here is because
+          // handle_broadcast_elementwise handles this selected_optimized_path
+          // option a bit differently. This should really be resolved in
+          // handle_broadcast_elementwise. However, the current blocker is that
+          // handle_broadcast_elementwise tries to be agnostic of op. This
+          // should be fixed, likely by moving lambda creation to
+          // handle_broadcast_elementwise and it be aware of which op is being
+          // executed.
+          auto add_lambda = [&alpha_val_vec](auto x, auto y) {
+            return y + alpha_val_vec * x;
+          };
+          return torch::executor::handle_broadcast_elementwise<CTYPE>(
+              ctx, add_lambda, a, b, out, selected_optimized_path, alpha);
+        } else {
+          auto add_lambda = [&alpha_val_vec](auto x, auto y) {
+            return x + alpha_val_vec * y;
+          };
+          return torch::executor::handle_broadcast_elementwise<CTYPE>(
+              ctx, add_lambda, a, b, out, selected_optimized_path, alpha);
+        }
+      }
+    });
```

Comment:
I think you can just select the lambdas based on is_sub rather than duplicating the rest of the code under this `if constexpr`: https://godbolt.org/z/Esdz1exKj

Reply:
thats a good suggestion. I tried doing the same in a different way which didnt quite work, but i can try out your suggestion